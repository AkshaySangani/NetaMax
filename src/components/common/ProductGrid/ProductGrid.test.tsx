import { render, cleanup } from "@testing-library/react";
import { ProductGrid } from "./ProductGrid";
import { IProductGridProps } from "./IProductGridProps";

describe("ProductGrid component", () => {
  let mock: IProductGridProps;

  beforeEach(() => {
    mock = {
      products: [
        {
          id: "xxxxx",
          name: "Testing product",
          sku: "xxxxx",
          images: [],
          createdOnUtc: new Date(),
          updatedOnUtc: new Date(),
          seoFilename: "xxxxx",
          price: "4.04",
          oldPrice: "4.04",
          orderMinimumQuantity: 1,
          orderMaximumQuantity: 2,
          categoryId: "xxxxx",
          stockQuantity: 10,
          manageInventoryMethodId: 1,
          fullDescription: "Testing Description",
        },
        {
          id: "xxxxx111",
          name: "Testing product1",
          sku: "xxxxx111",
          images: [],
          createdOnUtc: new Date(),
          updatedOnUtc: new Date(),
          seoFilename: "xxxxx1",
          price: "5.04",
          oldPrice: "5.04",
          orderMinimumQuantity: 3,
          orderMaximumQuantity: 4,
          categoryId: "xxxxx1",
          stockQuantity: 11,
          manageInventoryMethodId: 2,
          fullDescription: "Testing Description1",
        },
      ],
      onProductClick: jest.fn(),
      addToCart: jest.fn(),
      removeFromCart: jest.fn(),
      getQtyOnCart: jest.fn(),
    };
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test("ProductGrid should be render correctly", () => {
    const { container } = render(<ProductGrid {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("number of rendered elements match with the length of the testElements array", () => {
    const { queryAllByTestId } = render(<ProductGrid {...mock} />);
    expect(queryAllByTestId("product-container")).toHaveLength(mock.products.length);
  });
});
