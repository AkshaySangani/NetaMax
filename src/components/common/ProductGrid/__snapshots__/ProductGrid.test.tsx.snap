// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductGrid component ProductGrid should be render correctly 1`] = `
<div>
  <div
    class="css-1ymg3tg"
  >
    <div
      class="css-0"
      data-testid="product-container"
    >
      <div
        class="css-121pkih"
      >
        <div
          class="css-lfeor1"
          data-testid="product-button"
        >
          <div
            class="css-0"
          >
            <div
              class="css-j71k6g"
            >
              <p
                class="chakra-text css-w8ktmk"
                data-testid="product-tag"
              />
            </div>
          </div>
          <img
            alt="Testing product"
            class="chakra-image__placeholder css-2y8fku"
            draggable="false"
            src="/assets/images/noimg.png"
          />
        </div>
        <div
          class="css-8zy5qf"
        >
          <div
            class="css-g89h0y"
          >
            <p
              class="chakra-text css-1lgzimr"
            >
              Testing product
            </p>
          </div>
          <div
            class="css-70qvj9"
            data-testid="discount-box"
          >
            <span
              class="chakra-text css-d3r09a"
            >
              $4.04
            </span>
          </div>
          <div
            class="css-0"
          />
        </div>
      </div>
      <div
        class="css-k008qs"
      >
        <button
          class="chakra-button css-yzpepq"
          data-testid="add-to-card-button"
          type="button"
        >
          Añadir
        </button>
      </div>
    </div>
    <div
      class="css-0"
      data-testid="product-container"
    >
      <div
        class="css-121pkih"
      >
        <div
          class="css-lfeor1"
          data-testid="product-button"
        >
          <div
            class="css-0"
          >
            <div
              class="css-j71k6g"
            >
              <p
                class="chakra-text css-w8ktmk"
                data-testid="product-tag"
              />
            </div>
          </div>
          <img
            alt="Testing product1"
            class="chakra-image__placeholder css-2y8fku"
            draggable="false"
            src="/assets/images/noimg.png"
          />
        </div>
        <div
          class="css-8zy5qf"
        >
          <div
            class="css-g89h0y"
          >
            <p
              class="chakra-text css-1lgzimr"
            >
              Testing product1
            </p>
          </div>
          <div
            class="css-70qvj9"
            data-testid="discount-box"
          >
            <span
              class="chakra-text css-d3r09a"
            >
              $5.04
            </span>
          </div>
          <div
            class="css-0"
          />
        </div>
      </div>
      <div
        class="css-k008qs"
      >
        <button
          class="chakra-button css-yzpepq"
          data-testid="add-to-card-button"
          type="button"
        >
          Añadir
        </button>
      </div>
    </div>
  </div>
</div>
`;
